src/theme/hooks/useTheme.ts

```diff
@@ -125,6 +72,55 @@ export const useSubscribeDefaultAppTheme = () => {
       }
     }
   }, [touched, setTheme])
+
+  return {
+    theme,
+    set: (value: ThemeConfig) => {
+      setTheme((prev) => ({ ...prev, theme: value, touched: true }))
+    },
+    setDarkTheme(enabled: boolean) {
+      setTheme((prev) => {
+        const fallbackTheme =
+          prev.theme !== getDarkTheme(prev.themes)
+            ? prev.theme
+            : getLightTheme(prev.themes)
+        return {
+          ...prev,
+          theme: enabled ? getDarkTheme(prev.themes) : fallbackTheme,
+          touched: true
+        }
+      })
+    },
+    setLightTheme(enabled: boolean) {
+      setTheme((prev) => {
+        const fallbackTheme =
+          prev.theme !== getLightTheme(prev.themes)
```

Comment:
@neural-machine how do we know if a theme is light or dark?

Reply:
@neural-machine think it would be best if we were to have a new field `kind` or `type` to indicate if the type of a given theme is light or dark accordingly. I would also look into ways of defining a default light/dark themes by providing a flag and applying them when relying on the OS setting.